vars/pluginUtils.groovy

```diff
@@ -0,0 +1,88 @@
+def runPluginTests(plugins, envName, markerName, slackChannel, slackUrl) {
+    def lockName = "${env.JOB_NAME}-${envName}"
+    def pluginNames = plugins.keySet()
+
+    if (markerName != "") {
+        lockName = lockName + "-${markerName}"
+    }
+
+    lock(lockName) {
+        timeout(time: 150, unit: "MINUTES") {
+            results = pipelineUtils.runParallel(prepareStages(pluginNames, envName, markerName))
+        }
+    }
+
+    node {
+        writeFile(
+            file: "ibutsu.html",
+            text: (
+                "<a href=\"https://ibutsu.cloud.paas.psi.redhat.com/results?" +
+                "metadata.jenkins.build_number[eq]=${env.BUILD_NUMBER}&metadata.jenkins.job_name[eq]=" +
+                "${env.JOB_NAME}\">Ibutsu results</a>"
+            )
+        )
+
+        archiveArtifacts "ibutsu.html"
+
+        if (results['failed'].size() > 0) { 
+            def failedPlugins = plugins.subMap(results['failed'])
+            slackUtils.sendTestsFailedSlackMsg(failedPlugins, slackChannel, slackUrl)
+            error("Tests did not pass")
+        }
+    }
+}
+
+def prepareStages(pluginNames, jobName, envName, markerName) {
+    def stages = [:]
+    for (pluginName in pluginNames) {
+        // according to https://jenkins.io/doc/pipeline/examples/#parallel-multiple-nodes
+        def plugin = pluginName
+        stages[plugin] = {
+            def markerOption = ""
+            if (markerName != "") {
+                markerOption = "-m ${markerName}"
```

Comment:
Can you borrow the logic from here to allow markerName to be a list OR a string?

https://github.com/RedHatInsights/insights-pipeline-lib/blob/master/vars/execSmokeTest.groovy#L173-L175

https://github.com/RedHatInsights/insights-pipeline-lib/blob/master/vars/execSmokeTest.groovy#L184